### ISSUE TITLE
Paddle 训练效率慢问题

### ISSUE THREAD

**Issue by zhangyong15** [PaddlePaddle/Paddle#3616]
输入为dense vector， 162维，数据量约2.5T左右，100个MPI节点（内存估计60G），平均每轮需要2h。这个是不是不大符合预期啊，麻烦有提供加速的参数配置么？
另：每轮都需要重新下载数据么？

网络配置：
batch_size = 1000 if not is_predict else 1

embedding_layer = data_layer(name='embedding_layer', size=162)
fc_layer1 = fc_layer(name='fc_layer1', input=embedding_layer, size=128, act=ReluActivation())
fc_layer2 = fc_layer(name='fc_layer2', input=fc_layer1, size=64, act=ReluActivation())
fc_layer3 = fc_layer(name='fc_layer3', input=fc_layer2, size=32, act=ReluActivation())

prediction = fc_layer(
    name="prediction",
    input=fc_layer3,
    size=2,
    act=SoftmaxActivation())

if not is_predict:
    label = data_layer(name="label", size=2)
    cost= cross_entropy(prediction, label)
    outputs(cost)

dataprovider.py
@provider(
    init_hook=hook,
    should_shuffle=True,
    pool_size=40000,
    min_pool_size=0)
def process(settings, file_name):

**Comment by zhangyong15** [comment]
尝试使用：cache=CacheType.CACHE_PASS_IN_MEM优化
